lib/StaticAnalyzer/Core/BugReporter.cpp

```diff
@@ -1267,7 +1267,30 @@ static void AddInternalInfoToPathDiagnosticPiece(
   if (Pieces.empty())
     return;
 
-  // TODO: Add InterInfos to the first PathDiagnosticPiece in \p Pieces.
+  InternalInfoMap &Map = Pieces.front()->getInternalInfos();
+  auto &Opts = PDB.getBugReporter().getAnalyzerOptions();
+  ProgramStateRef State = N->getState();
+
+  if (Opts.isInternalInfoEnabled("state")) {
+    StateInfo &SI = Map.get<StateInfo>();
+    if (SI.isEmpty())
+      SI.addStateInfoFromState(State);
+  }
+
+  if (Opts.isInternalInfoEnabled("constraints")) {
+    ConstraintInfo &CurrC = Map.get<ConstraintInfo>();
+    CurrC.addConstraintsFromState(State);
+
+    // Sadly, ProgramState objects tend not to contain information from previous
+    // states, so we'll cascade them to the rest of the PathDiagnosticPiece
+    // objects within \p Pieces.
+    for (PathPieces::iterator I = ++Pieces.begin(),
```

Comment:
Can't we use a range based `for`?

Reply:
The range starts from element 1 as opposed to element 0. (`I = `**`++`**`Pieces.begin()`)